<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="input/detected_objects" default="/perception/object_recognition/detection/objects"/>
  <arg name="output/objects" default="/perception/object_recognition/detection/validation/occupancy_grid_based/objects"/>

  <arg name="input/laserscan" default="/perception/occupancy_grid_map/virtual_scan/laserscan"/>
  <arg name="input/occupancy_grid_map" default="/perception/object_recognition/detection/validation/occupancy_grid_based/single_frame_occupancy_grid_map"/>

  <node pkg="probabilistic_occupancy_grid_map" exec="laserscan_based_occupancy_grid_map_node" name="single_frame_laserscan_occupancy_grid_map" output="screen">
    <remap from="~/input/laserscan" to="$(var input/laserscan)"/>
    <remap from="~/output/occupancy_grid_map" to="$(var input/occupancy_grid_map)"/>
    <param name="input_obstacle_pointcloud" value="false"/>
    <param name="input_obstacle_and_raw_pointcloud" value="false"/>
    <param name="map_resolution" value="0.2"/>
    <param name="map_length" value="200.0"/>
    <param name="map_width" value="20.0"/>
    <param name="map_frame" value="base_link"/>
    <param name="enable_single_frame_mode" value="true"/>
  </node>

  <node pkg="detected_object_validation" exec="occupancy_grid_based_validator_node" name="occupancy_grid_based_validator_node" output="screen">
    <remap from="~/input/detected_objects" to="$(var input/detected_objects)"/>
    <remap from="~/input/occupancy_grid_map" to="$(var input/occupancy_grid_map)"/>
    <remap from="~/output/objects" to="$(var output/objects)"/>
    <param name="map_frame" value="base_link"/>
    <param name="enable_debug" value="false"/>
  </node>
</launch>
